src/libsyntax/feature_gate.rs

```diff
@@ -530,6 +530,7 @@ pub const BUILTIN_ATTRIBUTES: &'static [(&'static str, AttributeType, AttributeG
     ("derive", Normal, Ungated),
     ("should_panic", Normal, Ungated),
     ("ignore", Normal, Ungated),
+    ("allow_fail", Normal, Ungated),
```

Comment:
Yes, I got sidetracked and haven't been able to go back and add the feature gate yet. I will push that change asap